src/Native/Unix/System.Native/pal_console.cpp

```diff
@@ -430,7 +431,11 @@ static bool InitializeSignalHandling()
     // thread.  We can't do anything interesting in the signal handler,
     // so we instead send a message to another thread that'll do
     // the handling work.
+#if HAVE_PIPE2
+    if (pipe2(g_signalPipe, O_CLOEXEC) != 0)
+#else
     if (pipe(g_signalPipe) != 0)
+#endif
```

Comment:
Can you just use SystemNative_Pipe instead?  It already uses pipe2 if available or else uses pipe and then fcntl with FD_CLOEXEC.